pkg/controller/deployment/deployment_controller.go

```diff
@@ -85,6 +85,8 @@ type DeploymentController struct {
 
 	// Deployments that need to be synced
 	queue workqueue.RateLimitingInterface
+	// Deployments that need to be checked for progress.
+	progressQueue workqueue.RateLimitingInterface
```

Comment:
I didn't want to add the additional requeues in the main queue because syncing stuck deployments is not that easy. I can only make a good guess about the last time the controller will update the status of a deployment in the middle of a rollout. In a pathological case where a deployment runs many pods, the controller may end up resyncing the deployment many times where it won't need to update the deployment status. Think that every update will produce another resync where nothing may need to be updated. So I opted for a different queue instead of having a chance to ddos the main one. Also the progress check (what would need to be abstracted by the current code and run before anything else for every deployment) depends on resolving overlapping deployments. In order to resolve overlapping deployments, we do a List of all the deployments in the namespace. By having something else just run the progress checks, we don't List deployments that often.